keps/sig-cluster-lifecycle/kubeadm/4214-separate-super-user-kubeconfig/README.md

```diff
@@ -0,0 +1,724 @@
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+During the initial control plane node creation (`kubeadm init`) an `admin.conf` file is generated.
+This file currently contains a
+[`cluster-admin`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)
+credential that is bound to the `system:masters` group. Create two separate files instead -
+`admin.conf` containing a regular Kubernetes cluster-admin credential and a `super-admin.conf`
+containing a cluster-admin credential bound to the `system:masters` group.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Binding an admin credential to the`system:masters` group means it can
+[bypass RBAC](https://github.com/kubernetes/kubeadm/issues/2414#issue-836108390) - i.e.
+its permissions cannot be removed.
+At the time of writing this KEP, Kubernetes does not support
+[certificate revocation](https://github.com/kubernetes/kubernetes/issues/18982). This means
+the only way to revoke access of an admin credential bound to the `system:masters` group is to
+[rotate the certificate authority](https://kubernetes.io/docs/tasks/tls/manual-rotation-of-ca-certificates/)
+of this cluster.
+
+A general purpose `admin.conf` must be created that does not bind to `system:masters`.
+This credential can be shared by kubeadm deployed control plane nodes. In case this `admin.conf`
+credential is compromised, its permission must be revocable with RBAC.
+
+A separate "break-glass", super-user credential can be managed in `super-admin.conf`.
+This credential can be used to restore the cluster to a normal state in case of disruptive
+admin user activity.
+
+Both files must not be shared to new admin users and instead new admin credentials should
+be signed with the `kubeadm kubeconfig user` command.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Start managing a separate `super-admin.conf` that contains a super-user credential.
+- Continue using a file named `admin.conf` for all its current kubeadm uses today.
+- Improve the kubeadm documentation at the k8s.io website for creating
+additional admin users.
```

Comment:
Add it to `certs check-expiration` list?
- kubeadm  certs renew `super-admin.conf`

This is mentioned in the proposal. So we may not add this here.

Reply:
it's an implementation detail, so yes no need to mention it in goals.

it will definitely be possible to renew, but i was thinking about rne-using the existing "... review admin.conf` for both admin.conf and super-admin.conf. maybe that is not a good idea, so maybe a separate subcommand is better.